.pre-commit-config.yaml

```diff
@@ -30,3 +30,21 @@ repos:
     rev: 23.7.0
     hooks:
       - id: black
+
+  - repo: https://github.com/gitleaks/gitleaks
+      rev: v8.16.3
+    hooks:
+      - id: gitleaks
+
+  - repo: https://github.com/jumanjihouse/pre-commit-hooks
+    rev: 3.0.0
+    hooks:
+      - id: shellcheck
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: v1.3.5
+    hooks:
+      - id: cpplint
```

Comment:
these checks are running locally, to my knowledge, not on github. Have you tried running them on our codebase?